crates/codegen/grammar/src/version_quality.rs

```diff
@@ -0,0 +1,18 @@
+use super::*;
+
+use semver::Version;
+#[derive(Clone, Debug, Copy, PartialEq, Eq, strum_macros::Display)]
+pub enum VersionQuality {
+    Enabled,
+    Disabled,
```

Comment:
question: What would be the difference between these:

- `Enabled` vs `Valid`
- `Disabled` vs `Deprecated` vs `Invalid`

How can one happen without the other?

Reply:
I'm asking about the difference between `Enabled` and `Valid`. Can something be valid but not enabled?
Also the difference between `Disabled`, `Deprecated`, and `Invalid`. Can something be disabled but not deprecated?

Reply:
Removed Invalid and Valid. They were indeed the same as Enabled and Disabled.

A Disabled keyword (e.g.) can be used as an identifier. But a deprecated one cannot, even though the construct it is part of is no longer valid.

Reply:
The latest changes no longer have `Disabled` or `Enabled`. What changed here? how are keywords annotated now?

Reply:
Introduced and Removed cf. slack comment